### PR TITLE
Fix S3 backend support in iRODS object store

### PR DESCRIPTION
**Pull request by kxk302**
## What did you do? 
- [describe the proposed changes]
- I am working with @ksuderman to enable Galaxy to access S3 storage (part of me helping the Anvil team). Used the S3 pluggin and tried uploading a file (or paste data) and found a couple of issues that this PR addresses.

## Why did you make this change?
(Cite Issue number OR provide rationalization of changes if no issue exists)
(If fixing a bug, please add any relevant error or traceback)
To enable Galaxy to access S3 storage.

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ X] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

## For UI Components
- [ ] I've included a screenshot of the changes

### CODE REVIEW

File: lib/galaxy/objectstore/irods.py [hunk 1]
```diff
@@ -419,18 +419,26 @@ def _push_to_irods(self, rel_path, source_file=None, from_string=None):
                 log.debug("Wanted to push file '%s' to iRODS collection '%s' but its size is 0; skipping.", source_file, rel_path)
                 return True
 
+            # Create sub-collection first
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:422]: Creating a sub-folder and a data object must happen in both if/else clause below. Moved these steps before if/else clause.

File: lib/galaxy/objectstore/irods.py [hunk 2]
```diff
@@ -419,18 +419,26 @@ def _push_to_irods(self, rel_path, source_file=None, from_string=None):
...
+
             if from_string:
-                data_obj = self.session.data_objects.create(data_object_path, self.resource, **options)
+                # Save 'from_string' as a file
                 with data_obj.open('w') as data_obj_fp:
                     data_obj_fp.write(from_string)
+
+                # Add file containing 'from_string' to the irods collection, since
+                # put() expects a file as input. Get file name from data object's 'desc' field
+                self.session.data_objects.put(data_obj.desc, collection_path + "/", **options)
```
Comment (kxk302) [inline lib/galaxy/objectstore/irods.py:435]: Must use put() to make code indifferent to storage type (S3, disk, etc). put() requires a file as input. We can use the data object file saved above (file path given by data object's 'desc' field)

**Comment by kxk302** [comment]
With this change, you can upload a file (or paste data) in irods for both when a disk storage and S3 storage is used.

**Review by mvdbeek — APPROVED** [review]
Thanks @kxk302 @ksuderman!

**Pull request merged**